lib/net.js

```diff
@@ -125,6 +125,10 @@ function normalizeArgs(args) {
   const arg0 = args[0];
   var options = {};
   if (typeof arg0 === 'object' && arg0 !== null) {
+    // Handle the case where the arguments were already normalized.
+    if (Array.isArray(arg0))
```

Comment:
I'm curious, wouldn't `Array.isArray` always be a safer choice? Or because we know how `arg0` is created in this specific case we can safely use `instanceof Array`?